Info-Orbs/src/widgets/webDataModel.cpp

```diff
@@ -18,10 +18,68 @@ String WebDataModel::getData() {
 void WebDataModel::setData(String data) {
     if (m_data != data) {
         m_data = data;
+        Serial.print(m_label);
+        Serial.print(": ");
+        Serial.print(m_data[0]);
+        Serial.print(" - ");
+        Serial.println(m_data);
+        if (data[0] == '[') {
+            Serial.println("JSON");
+            JsonDocument doc;
+            DeserializationError error = deserializeJson(doc, data);
+            setElementsCount(doc.size());
+            Serial.print("Count: " + String(m_elementsCount));
+            for (int i = 0; i < doc.size(); i++) {
+            // Serial.print(i);
+            Serial.print(" i: " + String(i));
+                WebDataElementModel element = WebDataElementModel();
+            Serial.print("  ");
+            Serial.print(i);
+                element.parseData(doc[i].as<JsonObject>());
+            Serial.print(" X: ");
+            Serial.print(element.getX());
+            Serial.print(" Y: ");
+            Serial.print(element.getY());
+            Serial.print(" Text: ");
+            Serial.print(element.getText());
+
+            Serial.print(" Type: ");
+            Serial.print(element.getType());
+            Serial.println("");
+            m_elements[i] = element;
+            }
+        }
         m_changed = true;
     }
 }
 
+WebDataElementModel& WebDataModel::getElements() {
+    return *m_elements;
+}
+
+WebDataElementModel WebDataModel::getElement(int index) {
+    return m_elements[index];
+}
+
+int32_t WebDataModel::getElementsCount() {
+    return m_elementsCount;
+}
+
+void WebDataModel::setElementsCount(int32_t count) {
+    if (m_elementsCount != count) {
+        m_elements = (WebDataElementModel *)malloc(sizeof(WebDataElementModel) * count);
```

Comment:
This does not run the constructor, so it might be safer to do `m_elements = new WebDataElementModel[count];`

Reply:
Additionally, if `m_elements` is already set here, you leak that so you need to `free(m_elements)` right now,
but if you go for the `new[]` then you need to `delete[] m_elements;`